src/azure-cli/azure/cli/command_modules/appservice/custom.py

```diff
@@ -4179,6 +4179,73 @@ def _check_zip_deployment_status(cmd, rg_name, name, deployment_status_url, slot
     return res_dict
 
 
+def _get_latest_deployment_id(cmd, rg_name, name, deployment_status_url, slot):
+    import requests
+    from azure.cli.core.util import should_disable_connection_verify
+
+    headers = get_scm_site_headers(cmd.cli_ctx, name, rg_name, slot)
+    total_trials = 30
```

Comment:
30 trials means, we wait for max one minute to get the deployment_id.. Isn't it too long wait time?

Reply:
First kudu will return a temp deployment id, so it might be fine to wait for ~1min to get the actual deployment id. Also, this is just a temporary workaround to get the deployment id until we have `/deploymentstatus/latest` ARM API available.